akka-actor/src/main/scala/akka/pattern/RetrySupport.scala

```diff
@@ -0,0 +1,48 @@
+/**
+ * Copyright (C) 2009-2017 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka.pattern
+
+import akka.actor.Scheduler
+
+import scala.concurrent.duration.FiniteDuration
+import scala.concurrent.{ ExecutionContext, Future }
+import scala.util.control.NonFatal
+
+/**
+ * This trait provides the retry utility function
+ */
+trait RetrySupport {
+
+  /**
+   * Given a function from Unit to Future, returns an internally retrying Future
+   * The first attempt will be made immediately, each subsequent attempt will be made after 'delay'
+   * A scheduler (eg context.system.scheduler) must be provided to delay each retry
+   * If attempts are exhausted the returned future is simply the result of invoking attempt
+   *
```

Comment:
"Note that the `attempt` function will be invoked on the given execution context for subsequent tries and therefore must be thread safe (not touch unsafe mutable state)." 

Or something along those lines, and also the same for the Java versions.